### PR TITLE
Remove the FVM Tracers code

### PR DESCRIPTION
**Pull request by mfdeakin-sandia**
Part of this is needed for the CAM integration, so I decided to get rid of them completely instead of waiting. I may have been overzealous in my removal, so a careful review would be appreciated.
This hasn't been fully tested yet, but currently appears to be BFB.

### CODE REVIEW

File: src/interp_movie_mod.F90 [hunk 1]
```diff
@@ -143,8 +138,6 @@ module interp_movie_mod
        1,2,3,5,0,  &   ! Q4
        1,2,3,5,0,  &   ! Q5
        1,2,5,0,0,  &   ! psC
```
Comment (oksanaguba) [inline src/interp_movie_mod.F90]: below line with psC is removed , in attributes, but not here?

File: src/preqx/prim_advection_mod.F90 [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 module prim_advection_mod
 #ifndef USE_KOKKOS_KERNELS
-  use prim_advection_mod_base, only: Prim_Advec_Tracers_remap_ALE, prim_advec_tracers_fvm, vertical_remap_interface, Prim_Advec_Tracers_remap, Prim_Advec_Tracers_remap_rk2
```
Comment (oksanaguba) [inline src/preqx/prim_advection_mod.F90]: Andrew  will address this better, but i think _ALE routine is in leap tracers...

Comment (ambrad) [reply src/preqx/prim_advection_mod.F90]: It's OK to get ride of these. A lot has changed in the SL code between XX and E3SM repos. I'll eventually add back what is needed.

File: src/share/cxx/prim_driver.cpp [hunk 3]
```diff
@@ -70,12 +70,7 @@ HybridVCoord  Errors::runtime_abort("CAM forcing not yet availble in C++.\n"
                         Errors::err_not_implemented);
   // call TimeLevel_Qdp(tl, qsplit, n0_qdp)
 
-  // if (ftype==0) then
-  //   call t_startf("ApplyCAMForcing")
-  //   call ApplyCAMForcing(elem, fvm, hvcoord,tl%n0,n0_qdp, dt_remap,nets,nete)
-  //   call t_stopf("ApplyCAMForcing")
-
-  // elseif (ftype==2) then
+  // if (ftype==2) then
```
Comment (oksanaguba) [inline src/share/cxx/prim_driver.cpp]: harmless but i wonder why you wanted to touch this, there will be ftype0 and ftype2 when CAM is on.

Comment (mfdeakin-sandia) [reply src/share/cxx/prim_driver.cpp]: oops, it's all commented; I just meant to get rid of the commented code entirely

**Comment by ambrad** [comment]
If you continue to have enthusiasm for removing code, you could also rm all the implicit-related code, principally:
```
implicit_mod
prim_implicit_mod
prim_si_*
solver_*
```
XX will never implement these, so it's untested code in this repo.

**Pull request merged**